### PR TITLE
Add search bar

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR adds search bar (without pagination).

### Results

Kapture 2020-03-04 at 15 09 06

### Testing strategy

- [x] I ensured my code is tested: unit and e2e
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

**Comment by VincentCATILLON** [comment]
Missing things:
- e2e test
- UX enhancements (clear results on field clear, back)
- infinite scroll

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "start:android:release": "npm run start:android -- --variant=release",
     "prebuild:ios:cmd": "npm run -s xcode:check",
     "build:ios:cmd": "cd ios && export RCT_NO_LAUNCH_PACKAGER=1 && xcodebuild -workspace Coorpacademy.xcworkspace -scheme Coorpacademy -derivedDataPath build",
-    "build:iphonesimulator:debug": "REACT_NATIVE_FLAVOR=E2E npm run build:ios:cmd -- -configuration Debug -sdk iphonesimulator | xcpretty",
```
Comment (VincentCATILLON) [inline package.json]: should not be removed

File: src/components/catalog-items.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+|};
+
+class CatalogItems extends React.PureComponent<Props> {
+  props: Props;
+
+  keyExtractor = (item: DisciplineCard | ChapterCard | EmptyCard, index: number) => {
+    const {testID = 'catalog-items'} = this.props;
+    const suffix =
+      // $FlowFixMe union type
+      (item && item.universalRef && item.universalRef.replace(/_/g, '-')) || `${index}-placeholder`;
```
Comment (esa-coorp) [inline src/components/catalog-items.js]: pourquoi on fait cela?

File: src/layer/data/cards.js [hunk 3]
```diff
@@ -321,6 +322,22 @@ export const fetchCards = async (
...
+  limit: number
+) => fetchCards(token, host, section.endpoint, offset, limit, {...section.query});
+
+export const fetchSearchCards = (
+  token: string,
+  host: string,
+  search: string,
+  offset: number,
+  limit: number
+) => fetchCards(token, host, '/api/v2/contents', offset, limit, {fullText: search});
```
Comment (esa-coorp) [inline src/layer/data/cards.js]: il faut exclure les adaptives ...

File: jest.config.unit.js [hunk 4]
```diff
@@ -33,13 +33,13 @@ module.exports = {
     },
     'src/containers': {
       // @todo increase this value
-      branches: 50,
+      branches: 49,
```
[the remark this answers is not in the document]
Comment (VincentCATILLON) [reply jest.config.unit.js]: rollbacked

File: src/components/catalog-items.js (hunk 2 above)
Comment (VincentCATILLON) [reply src/components/catalog-items.js]: to have unique key on `FlatList` component and have something in [kebab case](https://en.wiktionary.org/wiki/kebab_case)

File: src/layer/data/cards.js (hunk 3 above)
Comment (VincentCATILLON) [reply src/layer/data/cards.js]: it's already excluded in the `fetchCards` function [there](https://github.com/CoorpAcademy/mobile/pull/475/files#diff-1ef030f59c211e3ad65b6e4f8a658f7eR302)

File: src/components/catalog-items.js (hunk 2 above)
Comment (esa-coorp) [reply src/components/catalog-items.js]: `_.kebabCase`

File: __e2e__/utils.js [hunk 5]
```diff
@@ -81,6 +88,19 @@ export const tapCardOnSection = async (testID: string, index: number) => {
...
 };
 
+export const tapCardOnSearch = async (index: number) => {
+  const {width} = getDeviceDimensions();
+  const columns = 2;
+  const offsetY = ITEM_HEIGHT * (parseInt(index / columns, 10) - 1) + 1; // Scroll amount must be positive and greater than zero
+  await waitForExist('catalog-search-items');
+  await element(by.id('catalog-search-items')).scrollTo('top');
+  await element(by.id('catalog-search-items')).scroll(offsetY, 'down');
+  await element(by.id('catalog-search-items')).tapAtPoint({
```
[the remark this answers is not in the document]
Comment (VincentCATILLON) [reply __e2e__/utils.js]: seen with you, i'll update it to calculate the `numColumns` and support more than 2 columns there even if we don't need it at this time

File: __e2e__/utils.js [hunk 6]
```diff
@@ -72,32 +72,42 @@ export const tap = async (testID: string) => {
...
   const {width} = getDeviceDimensions();
-  const columns = 2;
-  const offsetY = ITEM_HEIGHT * (parseInt(index / columns, 10) - 1) + 1; // Scroll amount must be positive and greater than zero
-  await waitForExist('catalog-search-items');
-  await element(by.id('catalog-search-items')).scrollTo('top');
-  await element(by.id('catalog-search-items')).scroll(offsetY, 'down');
-  await element(by.id('catalog-search-items')).tapAtPoint({
-    x: width / 2 - ITEM_WIDTH / 2 + ITEM_WIDTH * ((index - 1) % columns),
-    y: offsetY + ITEM_HEIGHT / 2
+  const columns = isVertical ? parseInt(width / ITEM_WIDTH, 10) : undefined;
```
Comment (esa-coorp) [inline __e2e__/utils.js]:
```suggestion
  const nbColumns = isVertical ? parseInt(width / ITEM_WIDTH, 10) : undefined;
```

File: __e2e__/utils.js [hunk 7]
```diff
@@ -72,32 +72,42 @@ export const tap = async (testID: string) => {
...
-//   });
-// };
-
-export const tapCardOnSection = async (testID: string, index: number) => {
-  const offsetX = ITEM_WIDTH * (index - 1) + 1; // Scroll amount must be positive and greater than zero
-  await waitForExist(testID);
-  await element(by.id(testID)).scrollTo('left');
-  await element(by.id(testID)).scroll(offsetX, 'right');
-  await element(by.id(testID)).tapAtPoint({x: offsetX + ITEM_HEIGHT / 2, y: ITEM_WIDTH / 2});
+const getCardOffset = (index: number, columns?: number): {x: number, y: number} => {
```
Comment (esa-coorp) [inline __e2e__/utils.js]:
```suggestion
const getCardOffset = (index: number, nbColumns?: number): {x: number, y: number} => {
```

**Pull request merged**